### DISCUSSION TITLE
PWA in offline mode

### DISCUSSION THREAD

**Discussion by digger73** [unoplatform/uno#13616] [Q&A]
Hi 

I'm pretty new to WASM, PWA thing. I have added a manifest to the WASM project and the PWA installs and runs fine. However, when running the PWA without the webserver being online  (dev environment for now), the PWA complains that there is no contact to the server.
What / how can I make the PWA show the initial main page even when no connection is available in UNO ?

**Comment by jeromelaban** [comment]
Thanks for the question. In order to make sure that your app can be run offline, you can use the [lighthouse tooling in Chrome](https://developer.chrome.com/docs/lighthouse/pwa/). It will tell you what's missing to be fully offline.

**Reply by digger73** [reply]
Thank you for the feedback, however I guess that there already must be some knowledge about what is missing in the default WASM for a PWA to run offline. If for example creating a Blazor PWA offline is supported out of the box as I understand it.
So what I'm trying to understand is what is needed / missing and how to handle this in the UNO WASM project.

**Reply by jeromelaban** [reply]
Thanks. You'll likely need to run the lighthouse reports after you make modifications to the app, but indeed, the app should be working by default in offline mode when PWA is enabled. 

We do not have those details at this time and if you can help adding information to https://github.com/unoplatform/uno/issues/13618 it will be greatly appreciated. 

Thanks!

**Reply by digger73** [reply]
I did run ligthhouse for a PWA and it shows green except showing: Manifest doesn't have a maskable icon.
And then the manual checks have to be done as well, which I haven't done yet. But still throws error when I disconnect the server.
